test/e2e/main/main_test.go

```diff
@@ -124,25 +124,83 @@ func TestNodeStartup(t *testing.T) {
+	// from Infinispan CR to pods
+	if pod.Labels["my-pod-label"] != ispn.Labels["my-pod-label"] {
+		panic("Infinispan CR labels haven't been propagated to pods")
+	}
+
+	// from operator environment
+	if tutils.RunLocalOperator == "TRUE" {
+		// running locally, labels are hardcoded and set by the testsuite
+		if pod.Labels["operator-pod-label"] != "operator-pod-value" ||
+			ispn.Labels["operator-pod-label"] != "operator-pod-value" {
+			panic("Infinispan CR labels haven't been propagated to pods")
+		}
+	} else {
+		// operator deployed on cluster, labels are set by the deployment
+		if !areOperatorLabelsPropagated(spec.Namespace, ispnv1.OperatorPodTargetLabelsEnvVarName, pod.Labels) {
+			panic("Operator labels haven't been propagated to pods")
+		}
 	}
+
 	svcList := &corev1.ServiceList{}
-	tutils.ExpectNoError(testKube.Kubernetes.ResourcesList(ispn.ObjectMeta.Namespace, labels.Set{}, svcList))
+	tutils.ExpectNoError(testKube.Kubernetes.ResourcesList(ispn.ObjectMeta.Namespace, map[string]string{"infinispan_cr": "test-node-startup"}, svcList))
 	if len(svcList.Items) == 0 {
 		panic("No services found for cluster")
 	}
 	for _, svc := range svcList.Items {
-		if svc.Labels["my-svc-label"] != ispn.Labels["my-svc-label"] ||
-			svc.Labels["operator-svc-label"] != "operator-svc-value" ||
-			ispn.Labels["operator-svc-label"] != "operator-svc-value" {
-			panic("Labels haven't been propagated to services")
+		// from Infinispan CR to service
+		if svc.Labels["my-svc-label"] != ispn.Labels["my-svc-label"] {
+			panic("Infinispan CR labels haven't been propagated to services")
+		}
+
+		// from operator environment
+		if tutils.RunLocalOperator == "TRUE" {
+			// running locally, labels are hardcoded and set by the testsuite
+			if svc.Labels["operator-svc-label"] != "operator-svc-value" ||
+				ispn.Labels["operator-svc-label"] != "operator-svc-value" {
+				panic("Labels haven't been propagated to services")
+			}
+		} else {
+			// operator deployed on cluster, labels are set by the deployment
+			if !areOperatorLabelsPropagated(spec.Namespace, ispnv1.OperatorTargetLabelsEnvVarName, svc.Labels) {
+				panic("Operator labels haven't been propagated to services")
+			}
 		}
 	}
 }
 
+// areOperatorLabelsPropagated helper function that read the labels from the infinispan operator pod
+// and match them with the labels map provided by the caller
+func areOperatorLabelsPropagated(namespace, varName string, labels map[string]string) bool {
+	podList := &corev1.PodList{}
+	tutils.ExpectNoError(testKube.Kubernetes.ResourcesList(namespace, map[string]string{"name": "infinispan-operator"}, podList))
+	if len(podList.Items) == 0 {
+		panic("Cannot get the Infinispan operator pod")
+	}
+	labelsAsString := ""
+	for _, item := range podList.Items[0].Spec.Containers[0].Env {
+		if item.Name == varName {
+			labelsAsString = item.Value
+		}
+	}
+	if labelsAsString == "" {
+		return true
+	}
+	opLabels := make(map[string]string)
```

Comment:
If we have set of defined labels expected to present on Operator OLM deployment maybe we could just assert that expected labels with expected values are present on the pods?

Reply:
there's no a known set of labels: i.e. community operator is shipped without any label.